packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -181,41 +194,43 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
     groupedAuthConnectionId?: string;
     socialLoginEmail?: string;
   }) {
-    try {
-      const {
-        idTokens,
-        authConnectionId,
-        groupedAuthConnectionId,
-        userId,
-        authConnection,
-        socialLoginEmail,
-      } = params;
-      const hashedIdTokenHexes = idTokens.map((idToken) => {
-        return remove0x(keccak256AndHexify(stringToBytes(idToken)));
-      });
-      const authenticationResult = await this.toprfClient.authenticate({
-        authConnectionId: groupedAuthConnectionId || authConnectionId,
-        userId,
-        idTokens: hashedIdTokenHexes,
-        groupedAuthConnectionParams: {
-          authConnectionId,
+    return await this.#withControllerLock(async () => {
```

Comment:
It's a bit hard to see which changes are because of adding the controller lock, and which are because of adding password syncing functionality. i think ideally this could have been split into 2 PRs. it's OK now, just mentioning it for next time.